docs/docker-compose-registry.yml

```diff
@@ -0,0 +1,96 @@
+version: '2'
+
+services:
+  redis:
+    restart: always
+    image: sameersbn/redis:latest
+    command:
+    - --loglevel warning
+    volumes:
+    - redis:/var/lib/redis:Z
+
+  postgresql:
+    restart: always
+    image: sameersbn/postgresql:9.5-3
+    volumes:
+    - postgresql:/var/lib/postgresql:Z
+    environment:
+    - DB_USER=gitlab
+    - DB_PASS=password
+    - DB_NAME=gitlabhq_production
+    - DB_EXTENSION=pg_trgm
+
+  gitlab:
+    restart: always
+    image: sameersbn/gitlab:8.12.7
+    volumes:
+    - gitlab-data:/home/git/data:Z
+    - gitlab-logs:/var/log/gitlab
+    - ./certs:/certs
+    depends_on:
+    - redis
+    - postgresql
+    ports:
+    - "80:80"
+    - "10022:22"
+    external_links:
+    - "registry:registry.example.com"
+    environment:
+    - DEBUG=false
+
+    - DB_ADAPTER=postgresql
+    - DB_HOST=postgresql
+    - DB_PORT=5432
+    - DB_USER=gitlab
+    - DB_PASS=password
+    - DB_NAME=gitlabhq_production
+
+    - REDIS_HOST=redis
+    - REDIS_PORT=6379
+
+    - GITLAB_HTTPS=false
+    - SSL_SELF_SIGNED=false
+
+    - GITLAB_HOST=gitlab.example.com
+    - GITLAB_PORT=80
+    - GITLAB_SSH_PORT=10022
+    - GITLAB_RELATIVE_URL_ROOT=
+    - GITLAB_SECRETS_DB_KEY_BASE=secret
+    - GITLAB_SECRETS_SECRET_KEY_BASE=secret
+    - GITLAB_SECRETS_OTP_KEY_BASE=secret
+
+    - GITLAB_REGISTRY_ENABLED=true
+    - GITLAB_REGISTRY_HOST=registry.example.com
+    - GITLAB_REGISTRY_PORT=5000
+    - GITLAB_REGISTRY_API_URL=https://registry.example.com:5000
+    - GITLAB_REGISTRY_CERT_PATH=/certs/registry-auth.crt
+    - GITLAB_REGISTRY_KEY_PATH=/certs/registry-auth.key
+
+  registry:
+    restart: always
+    image: registry:2.4.1
+    ports:
+    - "5000:5000"
+    volumes:
+    - registry-data:/var/lib/registry
+    - ./certs:/certs
+    external_links:
+    - "gitlab:gitlab.example.com"
+    environment:
+    - REGISTRY_LOG_LEVEL=info
+    - REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY=/registry
```

Comment:
Yes, that's correct. Thanks. I fixed it now.

The registry data is persistent by storing it in a docker volume.